Build fails on Node 23

On Node.js v23.9.0, all three of `main`, `next` and `3.5.3` fail to build with "module.exports is not a valid specifier". Everything builds fine on Node.js v22.14.0. 

Tested on both Arch Linux and macOS 15.3.1.

```sh
$ yarn build
 Building packages
index.mjs..............................................................✘ [ERROR] module.exports is not a valid specifier [plugin evaluate]

    node_modules/esbuild/lib/main.js:1277:21:
      1277 │         let result = await callback({
           ╵                      ^

    at file:///home/andre/sources/prettier/scripts/build/esbuild-plugins/evaluate.js:29:21
    at Array.map (<anonymous>)
    at file:///home/andre/sources/prettier/scripts/build/esbuild-plugins/evaluate.js:16:12
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async requestCallbacks.on-load (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1277:22)
    at async handleRequest (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:648:11)

  This error came from the "onLoad" callback registered here:

    node_modules/esbuild/lib/main.js:1151:20:
      1151 │       let promise = setup({
           ╵                     ^

    at setup (file:///home/andre/sources/prettier/scripts/build/esbuild-plugins/evaluate.js:13:13)
    at handlePlugins (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1151:21)
    at buildOrContextImpl (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:874:5)
    at Object.buildOrContext (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:700:5)
    at /home/andre/sources/prettier/node_modules/esbuild/lib/main.js:2030:15
    at new Promise (<anonymous>)
    at Object.build (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:2029:25)
    at Object.build (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1880:51)
    at Object.runEsbuild [as build] (file:///home/andre/sources/prettier/scripts/build/build-javascript-module.js:272:41)

  The plugin "evaluate" was triggered by this import

    src/index.js:133:35:
      133 │ export { default as version } from "./main/version.evaluate.cjs";
          ╵                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

   FAIL

Error: Build failed with 2 errors:
node_modules/esbuild/lib/main.js:1277:21: ERROR: [plugin: evaluate] module.exports is not a valid specifier
/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1327:32: ERROR: [plugin: shim-commonjs-objects] /home/andre/sources/prettier/dist/index.mjs not exists
    at failureErrorWithLog (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1477:15)
    at /home/andre/sources/prettier/node_modules/esbuild/lib/main.js:946:25
    at /home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1355:9
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}
node:internal/modules/run_main:104
    triggerUncaughtException(
    ^

Error: Build failed with 2 errors:
node_modules/esbuild/lib/main.js:1277:21: ERROR: [plugin: evaluate] module.exports is not a valid specifier
/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1327:32: ERROR: [plugin: shim-commonjs-objects] /home/andre/sources/prettier/dist/index.mjs not exists
    at failureErrorWithLog (/home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1477:15)
    at /home/andre/sources/prettier/node_modules/esbuild/lib/main.js:946:25
    at /home/andre/sources/prettier/node_modules/esbuild/lib/main.js:1355:9
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}

Node.js v23.9.0
```

--- a/scripts/build-website.js
+++ b/scripts/build-website.js
@@ -71,7 +71,7 @@ async function buildPlaygroundFiles() {
       continue;
     }
 
-    const { default: pluginModule } = await import(url.pathToFileURL(dist));
+    const pluginModule = await import(url.pathToFileURL(dist));
 
     const plugin = {
       name: path.basename(fileName, ".js"),

--- a/scripts/build/esbuild-plugins/evaluate.js
+++ b/scripts/build/esbuild-plugins/evaluate.js
@@ -1,8 +1,6 @@
+import url from "node:url";
 import { isValidIdentifier } from "@babel/types";
-import createEsmUtils from "esm-utils";
 import serialize from "serialize-javascript";
-
-const { importModule } = createEsmUtils(import.meta);
 
 export default function esbuildPluginEvaluate() {
   return {
@@ -11,8 +9,9 @@ export default function esbuildPluginEvaluate() {
       const { format } = build.initialOptions;
 
       build.onLoad({ filter: /\.evaluate\.c?js$/u }, async ({ path }) => {
-        const module = await importModule(path);
+        const module = await import(url.pathToFileURL(path));
         const text = Object.entries(module)
+          .filter(([specifier]) => specifier !== "module.exports")
           .map(([specifier, value]) => {
             const code =
               value instanceof RegExp

